CMakeLists.txt

```diff
@@ -180,3 +180,36 @@ if(USE_UPDATER)
   endif()
 
 endif()
+
+# Get the latest commit hash
+execute_process(
+        COMMAND git rev-parse --short HEAD
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE GIT_COMMIT_HASH
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(GIT_COMMIT_HASH)
+  STRING(TOUPPER ${GIT_COMMIT_HASH} GIT_COMMIT_HASH)
```

Comment:
I don't think we should uppercase it, let's keep it lowercase as elsewhere for consistency and better visual style.